app/src/main/java/com/bongtu/baekseo/presentation/home/schedule/ScheduleViewModel.kt

```diff
@@ -3,109 +3,100 @@ package com.bongtu.baekseo.presentation.home.schedule
 
     fun fetchScheduleEvent() {
-        // TODO: 전체 일정 조회 API 연동
         viewModelScope.launch {
-            updateScheduleUiState(
-                value = UiState.Success(
-                    persistentListOf(
-                        ScheduleEvent(
-                            eventId = "1",
-                            hostInfo = ScheduleHostInfo(
-                                hostName = "공승준",
-                                hostNickname = "초록승준",
-                            ),
-                            eventInfo = ScheduleEventInfo(
-                                eventCategory = EventType.WEDDING,
-                                relationship = RelationType.FRIEND,
-                                cost = 10000,
-                                eventDate = LocalDate.of(2025, 3, 11),
-                            ),
-                        ),
-                        ScheduleEvent(
-                            eventId = "2",
-                            hostInfo = ScheduleHostInfo(
-                                hostName = "김종명",
-                                hostNickname = "봉준호",
-                            ),
-                            eventInfo = ScheduleEventInfo(
-                                eventCategory = EventType.FIRST_BD,
-                                relationship = RelationType.NEIGHBOR,
-                                cost = 10000,
-                                eventDate = LocalDate.of(2025, 2, 11),
-                            ),
-                        ),
-                        ScheduleEvent(
-                            eventId = "3",
-                            hostInfo = ScheduleHostInfo(
-                                hostName = "김헤정",
-                                hostNickname = "메정",
-                            ),
-                            eventInfo = ScheduleEventInfo(
-                                eventCategory = EventType.BIRTHDAY,
-                                relationship = RelationType.ALUMNI,
-                                cost = 10000,
-                                eventDate = LocalDate.of(2025, 1, 11),
-                            ),
-                        ),
+            eventRepository.getScheduleEvents(
+                page = uiState.value.page,
+                category = uiState.value.eventCategoryType.label,
+            ).onSuccess { response ->
+                val newEvents = response.events
+                val updatedList =
+                    if (uiState.value.page == 0) newEvents else uiState.value.eventList + newEvents
+
+                _uiState.update { currentState ->
+                    currentState.copy(
+                        scheduleLoadState = if (updatedList.isEmpty()) {
+                            UiState.Empty
+                        } else {
+                            UiState.Success(
+                                PageScheduleEvent(
+                                    events = updatedList.toPersistentList(),
```

Comment:
p2: 여기 toPersistentList 를 한번만 호출해서 사용해도 될것 같아요
```suggestion
                val updatedImmutableList = updatedList.toPersistentList()
                _uiState.update { currentState ->
                    currentState.copy(
                        scheduleLoadState = if (updatedList.isEmpty()) {
                            UiState.Empty
                        } else {
                            UiState.Success(
                                PageScheduleEvent(
                                    events = updatedImmutableList,
```
아래에도 쓰이는것 같아서 바꿀수 있을것 같습니다 !